src/python/pants/engine/target.py

```diff
@@ -1278,11 +1278,8 @@ def _prefix_glob_with_address(self, glob: str) -> str:
         return str(PurePath(self.address.spec_path, glob))
 
     @final
-    def path_globs(self, files_not_found_behavior: FilesNotFoundBehavior) -> Optional[PathGlobs]:
-        globs = self.sanitized_raw_value
-        if globs is None:
-            return None
```

Comment:
This early return caused issues because `RelativeSourceFiles` will always be empty, and we short circuited before attempting codegen.

There was no reason to early return, other than "avoiding unnecessary work".